PROTO_tests/tests/bitops_test.py

```diff
@@ -0,0 +1,219 @@
+    @classmethod
+    def setup_class(cls):
+        cls.int_arr = ak.arange(10)
+        cls.uint_arr = ak.cast(cls.int_arr, ak.uint64)
+        cls.bigint_arr = ak.cast(cls.int_arr, ak.bigint)
+        cls.bigint_arr.max_bits = 64
+        cls.edge_cases = ak.array([-(2**63), -1, 2**63 - 1])
+        cls.edge_cases_uint = ak.cast(ak.array([-(2**63), -1, 2**63 - 1]), ak.uint64)
+        cls.edge_cases_bigint = ak.cast(cls.edge_cases_uint, ak.bigint)
+        cls.edge_cases_bigint.max_bits = 64
+
+    def test_popcount(self):
+        ans = [0, 1, 1, 2, 1, 2, 2, 3, 1, 2]
+        assert self.int_arr.popcount().to_list() == ans
+
+        ans_uint = np.array(ans, ak.uint64).tolist()
+        assert self.uint_arr.popcount().to_list() == ans_uint
+        assert self.bigint_arr.popcount().to_list() == ans_uint
+
+        # Edge case input
+        ans = [1, 64, 63]
+        assert ak.popcount(self.edge_cases).to_list() == ans
+
+        ans_uint = np.array(ans, ak.uint64).tolist()
+        assert ak.popcount(self.edge_cases_uint).to_list() == ans_uint
+        assert ak.popcount(self.edge_cases_bigint).to_list() == ans_uint
+
+    def test_parity(self):
+        ans = [0, 1, 1, 0, 1, 0, 0, 1, 1, 0]
+        assert self.int_arr.parity().to_list() == ans
+
+        ans_uint = np.array(ans, ak.uint64).tolist()
+        assert self.uint_arr.parity().to_list() == ans_uint
+        assert self.bigint_arr.parity().to_list() == ans_uint
+
+        ans = [1, 0, 1]
+        assert ak.parity(self.edge_cases).to_list() == ans
+
+        ans_uint = np.array(ans, ak.uint64).tolist()
+        assert ak.parity(self.edge_cases_uint).to_list() == ans_uint
+        assert ak.parity(self.edge_cases_bigint).to_list() == ans_uint
+
+    def test_clz(self):
+        ans = [64, 63, 62, 62, 61, 61, 61, 61, 60, 60]
+        assert self.int_arr.clz().to_list() == ans
+
+        ans_uint = np.array(ans, ak.uint64).tolist()
+        assert self.uint_arr.clz().to_list() == ans_uint
+        assert self.bigint_arr.clz().to_list() == ans_uint
+
+        ans = [0, 0, 1]
+        assert ak.clz(self.edge_cases).to_list() == ans
+
+        ans_uint = np.array(ans, ak.uint64).tolist()
+        assert ak.clz(self.edge_cases_uint).to_list() == ans_uint
+        assert ak.clz(self.edge_cases_bigint).to_list() == ans_uint
+
+    def test_ctz(self):
+        ans = [0, 0, 1, 0, 2, 0, 1, 0, 3, 0]
+        assert self.int_arr.ctz().to_list() == ans
+
+        ans_uint = np.array(ans, ak.uint64).tolist()
+        assert self.uint_arr.ctz().to_list() == ans_uint
+        assert self.bigint_arr.ctz().to_list() == ans_uint
+
+        ans = [63, 0, 0]
+        assert ak.ctz(self.edge_cases).to_list() == ans
+
+        ans_uint = np.array(ans, ak.uint64).tolist()
+        assert ak.ctz(self.edge_cases_uint).to_list() == ans_uint
+        assert ak.ctz(self.edge_cases_bigint).to_list() == ans_uint
+
+    @pytest.mark.parametrize("size", pytest.prob_size)
+    def test_bigint_bitops(self, size):
```

Comment:
since this is just a correctness test that shouldn't change based on the size of the array, I think I'm leaning towards not parameterizing on `size` for this one

Reply:
Removed